pageserver/src/tenant/secondary/scheduler.rs

```diff
@@ -19,6 +20,18 @@ use super::{CommandRequest, CommandResponse};
 const MAX_SCHEDULING_INTERVAL: Duration = Duration::from_secs(10);
 const MIN_SCHEDULING_INTERVAL: Duration = Duration::from_secs(1);
 
+/// Jitter a Duration by an integer percentage.  Returned values are uniform
+/// in the range 100-pct..100+pct (i.e. a 5% jitter is 5% either way: a ~10% range)
+pub(super) fn period_jitter(d: Duration, pct: u32) -> Duration {
+    rand::thread_rng().gen_range((d * (100 - pct)) / 100..(d * (100 + pct)) / 100)
+}
+
+/// When a periodic task first starts, it should wait for some time in the range 0..period, so
+/// that starting many such tasks at the same time spreads them across the time range.
+pub(super) fn period_warmup(period: Duration) -> Duration {
+    rand::thread_rng().gen_range(Duration::ZERO..period)
```

Comment:
This will panic if period == Duration::ZERO (if the range is empty).

Reply:
Good point - we already had that problem with the existing code, but since I'm moving these into helpers that might be used elsewhere in future, I've added a branch to handle zero inputs.